### ISSUE TITLE
Rule for all in tests fails 

### ISSUE THREAD

**Issue by fbi-ranger** [hercules-390/hyperion#173]
Dear all, 

Seems to be another issue in the make files: 

make[2]: Leaving directory '/local/sys1/src/git/hercules/hyperion/crypto'
Making all in tests
make[2]: Entering directory '/local/sys1/src/git/hercules/hyperion/tests'
make[2]: *** No rule to make target 'all'.  Stop.
make[2]: Leaving directory '/local/sys1/src/git/hercules/hyperion/tests'
Makefile:2250: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/local/sys1/src/git/hercules/hyperion'
Makefile:1660: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by apple601601601** [comment]
Same issue, guess add an empty target all in the makefile will do it.

**Comment by apple601601601** [comment]
My workaround:

# out-of-source build.  Something close to the follwing is also present
# in    the  Makefile   that is built by configure in the build directory.
# Any files built using this makefile should be considered source since
# they are for use on Hercules, and thus independent of build options.

# If    you  are  doing  out-of-source  build (which you should), you can
# modify  this  file and save it as Makefile.  As long as you don't git
# add  it,      all will be fine, except that git status will show the file
# (as it should).

# Clone SATK from
# git@github.com:s390guy/SATK.git

# This assumes that SATK is installed in parallel with hercules/hyperion

APATH:=../../../SATK/tools

%.core: %.asm
        ASMPATH=. MACLIB=mac ${APATH}/asma.py -l $*.list -i $@ $<

ALL:

**Comment by herc4mac** [comment]
> On 25 Dec 2016, at 06:18, Ping Z <notifications@github.com> wrote:
> 
> My workaround:
> 
> out-of-source build. Something close to the follwing is also present
> 

a friendly reminder to Mr.  Ping Z

emailing with that writing style is considered all over the net as shouting … 

we do not need that kind of behaviour around here

My best Regards

E.

**Comment by jphartmann** [comment]
Yes, out-of-source build is now a hard requirement on UNIX.  We were not aware that it caused in-source builds to fail as Florian reported, but there is another report that "make clear" fails similarly when done in-source and that there is no failure when the instructions in BUILDING.txt are observed.

Enrico:  The tracker does not take the huge font; it is what created the email that seems to have blown up.  Since the original posting by Florian has been removed in the process, it looks as if Florian and Mr. Ping Z are the same identity as seen by github.

**Comment by fbi-ranger** [comment]
After analyzing the differences in the tests directory compared to previous releases, I found that there is now a makefile. It seems that is from the tests for SoftFloat. Deleting the file after configure step makes the make all work again. So my question is, if this make file, which is also part of Makefile.am is required or could be deleted.

**Comment by jphartmann** [comment]
I see no reason at all for this file, but others insist that it remains for building a distribution.  

The Makefile.am has always been there; enabling the build of the makefile was an attempt at fixing the problem that creating a file in the tests directory causes a complete Hyperion rebuild on the next make.

If you or anyone else has any idea of how to cope with this auto#$* feature, please speak up.

Can you confirm that this problem remains for an out-of-source build?

**Comment by fbi-ranger** [comment]
That is correct Makefile.am does not do any harm. 
But isn't it that the makefile will be build when ./configure is run? Why do I need then the makefile in the git? To my understanding it simply interferes with build process.

**Comment by jphartmann** [comment]
Looks like it.   But I need to understand why adding a file to tests make everything rebuild.  Is there some general check for new files anywhere?  Inquiring minds want to know.

**Comment by fbi-ranger** [comment]
Thanks for removing the file.
Build is working again. 
The issue can be closed.
